tests/test_filtsmooth/test_particlefiltsmooth/test_particle_filter.py

```diff
@@ -0,0 +1,123 @@
+import numpy as np
+import pytest
+
+from probnum import filtsmooth, random_variables, statespace
+
+from ..filtsmooth_testcases import pendulum
+
+
+def test_effective_number_of_events():
+    weights = np.random.rand(10)
+    categ = random_variables.Categorical(
+        support=np.random.rand(10, 2), probabilities=weights / np.sum(weights)
+    )
+    ess = filtsmooth.effective_number_of_events(categ)
+    assert 0 < ess < 10
+
+
+#####################################
+# Test the RMSE on a pendulum example
+#####################################
+
+# Measmod style checks bootstrap and Gaussian proposals.
+all_importance_distributions = pytest.mark.parametrize("measmod_style", ["uk", "none"])
+
+# Resampling percentage threshold checks that
+# resampling is performed a) never, b) sometimes, c) always
+all_resampling_configurations = pytest.mark.parametrize(
+    "resampling_percentage_threshold", [-1.0, 0.1, 2.0]
+)
+
+
+@pytest.fixture
+def num_particles():
+    return 4
```

Comment:
How realistic is the usecase for huge numbers of particles? Because, spontaneously, I suspect that the algorithm (e.g. the estimator for the number of effective particles might get into numerical trouble), in this case. I have no practical experience, but maybe we would want to check when it starts to get difficult?

Reply:
Good question. From a theoretical perspective: the more the better. From a runtime perspective: I am not sure whether we can afford too many particles here. I have hope, though, that we can vectorize the implementation a bit, which would drastically improve the computation.